From 7100c7b97e0abffcfa99714989ea80902c794a9d Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Fri, 18 Oct 2024 16:05:51 +0800
Subject: [PATCH] fix no update UfmProvisioned property

If the PFR mailbox is not on I2C4 or the address is not 0x38 on the I2C bus,
it cannot retrieve the provision state and set UfmLocked, UfmProvisioned
and UfmSupport properties to false.

After getting the correct I2C bus and address from the entity-manager,
where UfmLocked, UfmProvisioned, and UfmSupport D-Bus properties do not update.

Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 service/src/pfr_mgr.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/service/src/pfr_mgr.cpp b/service/src/pfr_mgr.cpp
index d31c14f..65e7114 100644
--- a/service/src/pfr_mgr.cpp
+++ b/service/src/pfr_mgr.cpp
@@ -146,7 +146,7 @@ PfrConfig::PfrConfig(sdbusplus::asio::object_server& srv_,
 
     pfrCfgIface->register_property(ufmProvisionedStr, ufmProvisioned,
                                    // Override set
-                                   [this](const bool req, bool propertyValue) {
+                                   [this](const bool req, bool& propertyValue) {
                                        if (internalSet)
                                        {
                                            if (req != propertyValue)
@@ -161,7 +161,7 @@ PfrConfig::PfrConfig(sdbusplus::asio::object_server& srv_,
 
     pfrCfgIface->register_property(ufmLockedStr, ufmLocked,
                                    // Override set
-                                   [this](const bool req, bool propertyValue) {
+                                   [this](const bool req, bool& propertyValue) {
                                        if (internalSet)
                                        {
                                            if (req != propertyValue)
@@ -176,7 +176,7 @@ PfrConfig::PfrConfig(sdbusplus::asio::object_server& srv_,
 
     pfrCfgIface->register_property(ufmSupportStr, ufmSupport,
                                    // Override set
-                                   [this](const bool req, bool propertyValue) {
+                                   [this](const bool req, bool& propertyValue) {
                                        if (internalSet)
                                        {
                                            if (req != propertyValue)
